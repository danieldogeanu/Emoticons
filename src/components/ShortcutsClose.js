import React, { Component } from 'react';
import Icon from '../elements/Icon';
import ScreenReaderText from '../elements/ScreenReaderText';
import { animateCSS } from '../animate';

class ShortcutsClose extends Component {
	handleClick = () => {
		animateCSS(this.ShortcutsScreen, 'fadeOut');
		this.ShortcutsButton.classList.remove('hide');
		setTimeout(() => {
			this.ShortcutsScreen.classList.remove('show');
			this.ShortcutsScreen.classList.remove('animated', 'fadeIn', 'fadeOut');
		}, 290);
	}

	componentDidMount() {
		this.ShortcutsScreen = document.querySelector('.ShortcutsScreen');
		this.ShortcutsButton = document.querySelector('.ShortcutsButton');
	}

	render() {
		const shortcutsCloseText = 'Close Shortcuts Screen';
		return (
			<button className="ShortcutsClose"
				title={shortcutsCloseText}
				onClick={this.handleClick}>
				<ScreenReaderText text={shortcutsCloseText} />
				<Icon name="close" />
			</button>
		);
	}
}

export default ShortcutsClose;
