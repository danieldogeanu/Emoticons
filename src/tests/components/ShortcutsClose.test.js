import React from 'react';
import {render, fireEvent, cleanup, waitForDomChange} from '@testing-library/react';
import ShortcutsClose from '../../components/ShortcutsClose';

const btnTitle = 'Close Shortcuts Screen';

describe('ShortcutsClose Component', () => {

	afterEach(cleanup);

	it('renders shortcuts close button properly', () => {
		const {getByTitle, getByTestId, container} = render(<ShortcutsClose />);
		const renderedBtn = getByTitle(btnTitle);
		const renderedSRText = getByTestId('ScreenReaderText');
		const renderedIcon = container.querySelector('svg');

		expect(renderedBtn).toBeInTheDocument();
		expect(renderedBtn).toHaveClass('ShortcutsClose');
		expect(renderedBtn).toHaveAttribute('title', btnTitle);
		expect(renderedBtn).toContainElement(renderedSRText);
		expect(renderedBtn).toContainElement(renderedIcon);
		expect(renderedBtn).toMatchSnapshot();

		expect(renderedSRText).toHaveTextContent(btnTitle);
		expect(renderedIcon).toHaveClass('icon', 'close');
	});

	it('closes shortcuts screen', async () => {
		const {getByTitle, getByTestId} = render(
			<div data-testid="test-parent">
				<button className="ShortcutsButton hide"
					data-testid="ShortcutsButton" />
				<div className="ShortcutsScreen show animated fadeIn fadeOut"
					data-testid="ShortcutsScreen">
					<ShortcutsClose />
				</div>
			</div>
		);
		const shortcutsClose = getByTitle(btnTitle);
		const shortcutsButton = getByTestId('ShortcutsButton');
		const shortcutsScreen = getByTestId('ShortcutsScreen');

		fireEvent.click(shortcutsClose);
		await waitForDomChange();

		expect(shortcutsScreen).not.toHaveClass('show', 'animated', 'fadeIn', 'fadeOut');
		expect(shortcutsButton).not.toHaveClass('hide');
	});

});
