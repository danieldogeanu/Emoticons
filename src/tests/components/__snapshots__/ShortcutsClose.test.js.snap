// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShortcutsClose Component renders shortcuts close button properly 1`] = `
<button
  class="ShortcutsClose"
  title="Close Shortcuts Screen"
>
  <span
    class="show-for-screen-reader"
    data-testid="ScreenReaderText"
  >
    Close Shortcuts Screen
  </span>
  <svg
    class="icon close"
  >
    <use
      xlink:href="#close"
    />
  </svg>
</button>
`;
